### PR TITLE
RISC-V workflows: initial version

### PR DESCRIPTION
**Pull request by mshabunin**
waiting for https://github.com/opencv-infrastructure/opencv-gha-dockerfile/pull/20

### CODE REVIEW

**Comment by mshabunin** [comment]
@asmorkalov , where can I get the `print_system_information.sh` script which should be located in the `git_cache` folder in all CI jobs?

**Comment by opencv-alalek** [comment]
Just create empty / dummy print for testing.

[Example](https://github.com/opencv/opencv/actions/runs/6224002658/job/16891213155#step:3:21):
```
OS: Ubuntu 20.04 LTS
CPU:
	 model: Intel(R) Xeon(R) Silver 4[21](https://github.com/opencv/opencv/actions/runs/6224002658/job/16891213155#step:3:22)0 CPU @ 2.20GHz
	 number of CPU: 2
	 total cores: 40
GPU:
	 model: NVIDIA Quadro P2000
	 number of GPU: 1
RAM:
	 size: 64GB
```

File: .github/workflows/OCV-PR-4.x-RISCV.yaml [hunk 1]
```diff
@@ -41,6 +41,7 @@ jobs:
         - /mnt/cache/ci_cache/opencv:/home/ci/.ccache
         - /mnt/cache/binaries_cache:/home/ci/binaries_cache
         - /mnt/cache/dnn-models:/home/ci/dnn-models
+      options: '--user root'
```
Comment (asmorkalov) [inline .github/workflows/OCV-PR-4.x-RISCV.yaml:44]: Looks strange and unsafe,

**Comment by mshabunin** [comment]
It started to work, but I'm not sure why only one pipeline had run (Contrib). Should I create separate PR for main pipeline?

File: .github/workflows/OCV-PR-4.x-RISCV.yaml (hunk 1 above)
Comment (mshabunin) [reply .github/workflows/OCV-PR-4.x-RISCV.yaml]: It's OK, we use rootless docker install and it maps root user (0) in container to regular user on machine (e.g. 1001). I've moved this option to the image.

File: .github/workflows/OCV-PR-4.x-RISCV.yaml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+  PR_AUTHOR_FORK: ${{ github.event.pull_request.head.repo.full_name }}
+  SOURCE_BRANCH_NAME: ${{ github.head_ref }}
+  TARGET_BRANCH_NAME: ${{ github.base_ref }}
+  ANT_HOME: '/usr/share/ant'
+  GIT_CACHE_DOCKER: '/home/ci/git_cache'
+  PYTHONPATH: /home/ci/build/python_loader:$PYTHONPATH
+  OPENCV_TEST_DATA_PATH: '/home/ci/opencv_extra/testdata'
+  OPENCV_DOCKER_WORKDIR: '/home/ci/opencv'
+  DNN_MODELS: '/home/ci/dnn-models'
```
Comment (asmorkalov) [inline .github/workflows/OCV-PR-4.x-RISCV.yaml:26]: DNN data synchronization is redundant. DNN test is not executed.

File: .github/workflows/OCV-PR-4.x-RISCV.yaml [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+          echo "Merge opencv_extra with ${{ env.SOURCE_BRANCH_NAME }} branch"
+          cd $HOME/opencv_extra
+          git pull -v "https://github.com/${{ env.PR_AUTHOR }}/opencv_extra" "${{ env.SOURCE_BRANCH_NAME }}"
+        else
+          echo "No merge since ${{ env.PR_AUTHOR }}/opencv_extra does not have branch ${{ env.SOURCE_BRANCH_NAME }}"
+        fi
+    - name: Extra DNN models update
+      timeout-minutes: 60
+      run: |
+        DOWNLOAD_MODELS_FILE='download_models.py'
```
Comment (asmorkalov) [inline .github/workflows/OCV-PR-4.x-RISCV.yaml:97]: DNN data synchronization is redundant. DNN test is not executed.

**Review by asmorkalov — APPROVED** [review]

File: .github/workflows/OCV-PR-4.x-RISCV.yaml [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  SOURCE_BRANCH_NAME: ${{ github.head_ref }}
+  TARGET_BRANCH_NAME: ${{ github.base_ref }}
+  ANT_HOME: '/usr/share/ant'
+  GIT_CACHE_DOCKER: '/home/ci/git_cache'
+  PYTHONPATH: /home/ci/build/python_loader:$PYTHONPATH
+  OPENCV_TEST_DATA_PATH: '/home/ci/opencv_extra/testdata'
+  OPENCV_DOCKER_WORKDIR: '/home/ci/opencv'
+  DNN_MODELS: '/home/ci/dnn-models'
+  PARALLEL_JOBS: 16
+  TEST_RUNNER: '/opt/tools/qemu-riscv64 -cpu syntacore-scr9 -L /opt/riscv/sysroot'
```
Comment (asmorkalov) [inline .github/workflows/OCV-PR-4.x-RISCV.yaml:28]: path is incorrect. `/opt/tools/bin/qemu-riscv64`.

File: .github/workflows/OCV-PR-4.x-RISCV.yaml (hunk 2 above)
Comment (mshabunin) [reply .github/workflows/OCV-PR-4.x-RISCV.yaml]: Done.

File: .github/workflows/OCV-PR-4.x-RISCV.yaml (hunk 3 above)
Comment (mshabunin) [reply .github/workflows/OCV-PR-4.x-RISCV.yaml]: Done.

File: .github/workflows/OCV-PR-4.x-RISCV.yaml (hunk 4 above)
Comment (mshabunin) [reply .github/workflows/OCV-PR-4.x-RISCV.yaml]: Fixed, thank you.

**Comment by mshabunin** [comment]
@asmorkalov , @opencv-alalek , with new container (clang 16.0.6) all tests except one pass. I think this PR can be merged.

**Comment by opencv-alalek** [comment]
> Imgproc_Hist_Compare.accuracy

BTW, looks like it is a regression of OSoC refactoring:

https://pullrequest.opencv.org/buildbot/builders/4_x_riscv_rvv-clang-qemu-lin64?numbuilds=25

(failed since Sep 15)

**Review by asmorkalov — APPROVED** [review]
:+1:

**Review by opencv-alalek — APPROVED** [review]
Looks good to me :+1:

BTW, it makes sense to add DNN tests too (without large models at least).

**Pull request merged**